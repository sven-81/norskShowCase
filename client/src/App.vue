<script setup lang="ts">
import Master from './views/Master.vue'
</script>

<template>
  <Master />
</template>

<style>
@import '@/assets/css/norsk.css';
@import '@/assets/css/navi.css';
</style>