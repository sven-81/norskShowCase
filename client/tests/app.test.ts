import { beforeEach, describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import App from '@/App.vue'
import { createRouter, createWebHistory } from 'vue-router'
import { createPinia } from 'pinia'

const DummyView = {
  template: '<div>Dummy View</div>'
}

const routes = [
  {
    path: '/',
    component: DummyView
  }
]

const router = createRouter({
  history: createWebHistory(),
  routes
})

const pinia = createPinia()

let wrapper

beforeEach(async () => {
  import.meta.env.VITE_IMPRINT_MAIL = 'test@example.com'

  router.push('/')
  await router.isReady()
  wrapper = mount(App, {
    global: {
      plugins: [router, pinia]
    }
  })
})
describe('App.vue', () => {
  it('renders master layout without errors', async () => {
    expect(wrapper.html()).toContain('Dummy View')
    expect(wrapper.findComponent({ name: 'Master' }).exists()).toBe(true)
  })

  it('fits structure-snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can import mail', async () => {
    expect(wrapper.html()).toContain('mailto:test@example.com')
  })
})
