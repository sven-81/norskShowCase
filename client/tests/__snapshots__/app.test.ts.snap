// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > fits structure-snapshot 1`] = `
"<div class="wrapper">
  <header id="header" class="header">
    <h1><a href="/"><img alt="Norwegen Fahne" class="norway-flag" src="/src/assets/images/norway.svg">NORSK</a></h1>
    <h2>Norwegisch lernen</h2>
  </header>
  <nav class="menu">
    <ol>
      <li class=""><a href="">Trainer</a>
        <ol class="sub-menu">
          <li class=""><a href="/train/words">WÃ¶rter</a></li>
          <li class=""><a href="/train/verbs">Verben</a></li>
        </ol>
      </li>
      <!--v-if-->
      <li class=""><a href="/login">Login</a></li>
      <li class=""><a href="/register">Registrieren</a></li>
      <li class=""><a href="/imprint">Impressum</a></li>
    </ol>
  </nav>
  <!--v-if-->
  <div id="content" class="column row">
    <div>Dummy View</div>
  </div>
  <footer class="column">
    <div class="row">
      <div class="copyright icon-paw">
        <p><a href="mailto:test@example.com" title="Mail an test@example.com">test@example.com</a><br><span class="small">flag: publicdomainvectors.org</span></p>
      </div>
    </div>
  </footer>
</div>"
`;
